packages/edit-site/src/components/global-styles/variations-typography.js

```diff
@@ -0,0 +1,161 @@
+import { getFamilyPreviewStyle } from './font-library-modal/utils/preview-styles';
+import { useCurrentMergeThemeStyleVariationsWithUserConfig } from '../../hooks/use-theme-style-variations/use-theme-style-variations-by-property';
+import Subtitle from './subtitle';
+import Variation from './variation';
+
+const { GlobalStylesContext } = unlock( blockEditorPrivateApis );
+
+function getFontFamilyFromSetting( fontFamilies, setting ) {
+	if ( ! setting ) {
+		return null;
+	}
+
+	const fontFamilyVariable = setting.replace( 'var(', '' ).replace( ')', '' );
+	const fontFamilySlug = fontFamilyVariable?.split( '--' ).slice( -1 )[ 0 ];
+
+	return fontFamilies.find(
+		( fontFamily ) => fontFamily.slug === fontFamilySlug
+	);
+}
+
+const getFontFamilies = ( themeJson ) => {
+	const fontFamilies = themeJson?.settings?.typography?.fontFamilies?.theme; // TODO this could not be under theme.
+	const bodyFontFamilySetting = themeJson?.styles?.typography?.fontFamily;
+	const bodyFontFamily = getFontFamilyFromSetting(
+		fontFamilies,
+		bodyFontFamilySetting
+	);
+
+	const headingFontFamilySetting =
+		themeJson?.styles?.elements?.heading?.typography?.fontFamily;
+
+	let headingFontFamily;
+	if ( ! headingFontFamilySetting ) {
+		headingFontFamily = bodyFontFamily;
+	} else {
+		headingFontFamily = getFontFamilyFromSetting(
+			fontFamilies,
+			themeJson?.styles?.elements?.heading?.typography?.fontFamily
+		);
+	}
+
+	return [ bodyFontFamily, headingFontFamily ];
+};
+const getFontFamilyNames = ( themeJson ) => {
+	const [ bodyFontFamily, headingFontFamily ] = getFontFamilies( themeJson );
+	return [ bodyFontFamily?.name, headingFontFamily?.name ];
+};
+
+const TypePreview = ( { variation } ) => {
+	const { base } = useContext( GlobalStylesContext );
+	const [ bodyFontFamilies, headingFontFamilies ] = getFontFamilies(
+		mergeBaseAndUserConfigs( base, variation )
+	);
+	const bodyPreviewStyle = bodyFontFamilies
+		? getFamilyPreviewStyle( bodyFontFamilies )
+		: {};
+	const headingPreviewStyle = headingFontFamilies
+		? getFamilyPreviewStyle( headingFontFamilies )
+		: {};
+	return (
+		<motion.div
+			className="edit-site-global-styles-type-preview"
+			animate={ {
+				scale: 1,
+				opacity: 1,
+			} }
+			initial={ {
+				scale: 0.1,
+				opacity: 0,
+			} }
+			transition={ {
+				delay: 0.3,
+				type: 'tween',
+			} }
+		>
+			<span style={ headingPreviewStyle }>A</span>
+			<span style={ bodyPreviewStyle }>a</span>
+		</motion.div>
+	);
+};
+
+export default function TypographyVariations() {
+	const typographyVariations =
+		useCurrentMergeThemeStyleVariationsWithUserConfig( {
+			property: 'typography',
+			filter: ( variation ) =>
+				variation?.settings?.typography?.fontFamilies &&
+				Object.keys( variation?.settings?.typography?.fontFamilies )
+					.length,
+		} );
+
+	const { base } = useContext( GlobalStylesContext );
+	const uniqueTypographyVariations = [];
+	const uniqueTypographyNames = [];
+	const isDup = ( x, y ) => {
```

Comment:
If we're using `A` and `a` only, we could probably remove a bit of this code (`uniqueTypographyNames`, `isDup`, `getFontFamilyNames`)  E.g., 

```js
	const uniqueTypographyVariations = Object.values(
		typographyVariations?.reduce( ( acc, variation ) => {
			const [ bodyFontFamily, headingFontFamily ] = getFontFamilies(
				mergeBaseAndUserConfigs( base, variation )
			);
			if (
				bodyFontFamily?.name &&
				headingFontFamily?.name &&
				! acc[
					`${ bodyFontFamily?.name }:${ headingFontFamily?.name }`
				]
			) {
				acc[
					`${ bodyFontFamily?.name }:${ headingFontFamily?.name }`
				] = variation;
			}

			return acc;
		}, {} )
	);
```